tests/streams.test.cpp

```diff
@@ -479,4 +479,165 @@ boost::ut::suite multi_stream_test = []() {
     */
   };
 };
+
+boost::ut::suite stream_try_until_test = []() {
+  using namespace boost::ut;
+  using namespace std::literals;
+
+  "[try_until] normal usage"_test = []() {
+    // Setup
+    std::string_view digits_in_between = "abcd1234x---";
+    parse<std::uint32_t> parse_int;
+    auto getter = [&digits_in_between]() -> std::span<const hal::byte> {
+      return std::span<const hal::byte>(
+        reinterpret_cast<const hal::byte*>(digits_in_between.data()),
+        digits_in_between.size());
+    };
+
+    constexpr int timeout_call_limit = 10;
+    int counts = 0;
+    auto timeout_function = [&counts]() mutable -> status {
+      counts++;
+      if (counts >= timeout_call_limit) {
+        return hal::new_error(std::errc::timed_out);
+      }
+      return {};
+    };
+
+    // Exercise
+    auto result = try_until(parse_int, getter, timeout_function).value();
+
+    // Verify
+    expect(that % work_state::finished == result);
+  };
+
+  "[try_until] complete before timeout"_test = []() {
+    // Setup
+    std::string_view digits_in_between = "abcd1234x---";
+    parse<std::uint32_t> parse_int;
+    auto getter = [&digits_in_between]() -> std::span<const hal::byte> {
+      return std::span<const hal::byte>(
+        reinterpret_cast<const hal::byte*>(digits_in_between.data()),
+        digits_in_between.size());
+    };
+
+    auto timeout_function = []() mutable -> status { return {}; };
+
+    // Exercise
+    auto result = try_until(parse_int, getter, timeout_function).value();
+
+    // Verify
+    expect(that % work_state::finished == result);
+  };
+
+  "[try_until] stream in multiple parts"_test = []() {
+    // Setup
+    std::string_view stream_part1 = "abcd1";
+    std::string_view stream_part2 = "123";
+    std::string_view stream_part3 = "x---";
```

Comment:
```suggestion
    std::string_view stream_part3 = "4x---";
```
This way, the last part isn't ignored and is needed for the parsing to complete.